File: Data Structures & Algorithms/house-robber/submission-1.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        dp = [-1] * len(nums)
        dp[0], dp[1] = nums[0], max(nums[0], nums[1])

        def find_dp(val):
            if dp[val] == -1:
                dp[val] = max(find_dp(val - 1), find_dp(val - 2) + nums[val])
            return dp[val]
        
        return find_dp(len(nums) - 1)

File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::unordered_set<int> dupeSet;
        for (int num: nums) {
            if (dupeSet.count(num) == 1) {
                return true;
            }
            dupeSet.insert(num);
        }
        return false;
    }
};